share/ipld/namespace_data.go

```diff
@@ -0,0 +1,167 @@
+package ipld
+
+import (
+	"errors"
+	"fmt"
+	"sync/atomic"
+
+	"github.com/ipfs/go-cid"
+	ipld "github.com/ipfs/go-ipld-format"
+
+	"github.com/celestiaorg/nmt"
+	"github.com/celestiaorg/nmt/namespace"
+)
+
+// Option is the functional option that is applied to the NamespaceData instance
+// to configure data that needs to be stored.
+type Option func(*NamespaceData)
+
+// WithLeaves option specifies that leaves should be collected during retrieval.
+func WithLeaves() Option {
```

Comment:
maybe rename `WithLeaves` to `CollectLeaves` since leaves are not yet there and needs to be collected? Also `CollectLeaves` rename suggestion below.

Reply:
We are using `With` with options everywhere, so I would leave this pattern.